python/tank/deploy/tank_commands/unregister_folders.py

```diff
@@ -0,0 +1,311 @@
+        try:
+            pc.synchronize(log)
+        finally:
+            pc.close()
+
+        # get the filesystem location ids which are associated with the entity    
+        sg_data = self.tk.shotgun.find(path_cache.SHOTGUN_ENTITY, [[path_cache.SG_ENTITY_FIELD, "is", entity]])
+        sg_ids = [x["id"] for x in sg_data]
+        log.debug("The following path cache ids are linked to the entity: %s" % sg_ids)
+        
+        if len(sg_ids) == 0:
+            log.info("This entity does not have any folder associated!")
+            return []
+        
+        return self._unregister_filesystem_location_ids(sg_ids, log, prompt)                              
+        
+        
+    def _unregister_filesystem_location_ids(self, ids, log, prompt):
+        """
+        Performs the unregistration of a path from the path cache database.
+        Will recursively unregister any child items parented to the given
+        filesystem location id.
+        
+        :param ids: List of filesystem location ids to unregister
+        :param prompt: Should the user be presented with confirmation prompts?
+        :param log: Logging instance
+        :returns: List of dictionaries to represents the items that were unregistered.
+                  Each dictionary has keys path and sg_id.
+                  Note that the shotgun ids returned will refer to retired objects in 
+                  Shotgun rather than live ones.
+        """
+        # now use the path cache to get a list of all folders (recursively) that are
+        # linked up to the folders registered for this entity.
+        
+        paths = []
+        pc = path_cache.PathCache(self.tk)
+        try:
+            for sg_fs_id in ids:
+                paths.extend(pc.get_folder_tree_from_sg_id(sg_fs_id))
+        finally:
+            pc.close()
+                
+        log.info("")
+        log.info("The following folders will be unregistered:")
+        for p in paths:
+            log.info(" - %s" % p["path"])
+        
+        log.info("")
+        log.info("Proceeding would unregister the above paths from Toolkit's folder database. "
+                 "This will not alter any of the content in the file system, but once you have "
+                 "unregistered the paths, they will not be recognized by Shotgun.")
+        log.info("")
+        log.info("This is useful if you for example have renamed an Asset or Shot and want to "
+                 "move the data around on disk. In this case, start by unregistering the existing "
+                 "folder entries in Shotgun, then rename the Shot or Asset in Shotgun. Next, "
+                 "create new folders on disk using Toolkit's 'create folders' command and lastly "
+                 "move all your file system data across from the old location to the new location.")
+        log.info("")
+        if prompt:
+            val = raw_input("Proceed with unregister? (Yes/No) ? [Yes]: ")
```

Comment:
Just a "Proceed?" or "Proceed with unregistering the above folders?" would be better.